conan/cli/commands/graph.py

```diff
@@ -298,3 +300,110 @@ def graph_explain(conan_api, parser,  subparser, *args):
+
+    if path:
+        deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                         args.user, args.channel,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update,
+                                                         check_updates=args.check_updates,
+                                                         is_build_require=args.build_require)
+    else:
+        deps_graph = conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update,
+                                                         check_updates=args.check_updates)
+    print_graph_basic(deps_graph)
+
+    dependencies = deps_graph.nodes[1:]
+    resolved_ranges = list(deps_graph.resolved_ranges.keys())
+
+    latest_recipe = []
+
+    ConanOutput().title("Checking remotes")
+
+    # Data structure to solve the repeated dependencies in graph
+    dict_nodes = {}
+    for node in dependencies:
+        if node.name in dict_nodes:
+            dict_nodes[node.name].append(node)
+        else:
+            dict_nodes[node.name] = [node]
+
+    for node_name, node_list in dict_nodes.items():
+        ref_pattern = ListPattern(node_name, rrev=None, prev=None)
+        latest_remote_ref, found_remote, recipe_ref = None, None, None
+        for remote in remotes:
+            try:
+                remote_ref_list = conan_api.list.select(ref_pattern, package_query=None,
+                                                        remote=remote)
+                if not remote_ref_list.recipes:
+                    continue
+            except Exception:
+                ConanOutput(f"{node_name} not found in remote {remote.name}")
+                continue
+            str_latest_ref = list(remote_ref_list.recipes.keys())[-1]
+            recipe_ref = RecipeReference.loads(f"{str_latest_ref}")
+            if latest_remote_ref is None or latest_remote_ref < recipe_ref:
+                latest_remote_ref = recipe_ref
+                found_remote = remote
+        latest_cache_ref = max(node_list, key=lambda x: x.ref)
+        if recipe_ref is None:
+            continue
+        if len(node_list) != 1 or latest_cache_ref.ref < latest_remote_ref:
+            latest_recipe.append((latest_cache_ref, node_list, latest_remote_ref, found_remote))
+
+    return {node.name: {"current_versions": node_list,
+                        "latest_remote": {"ref": latest_remote_ref,
+                                          "remote": found_remote},
+                        "version_ranges": [r for r in resolved_ranges if r.name == node.name]}
+            for node, node_list, latest_remote_ref, found_remote in latest_recipe}
```

Comment:
Avoid a very complex definition of the return value, better prepare it earlier in a explicit step.